lib/syntax.ml

```diff
@@ -107,10 +107,17 @@ let exists_grouping columns =
   List.exists (function Expr (e,_) -> is_grouping e | All | AllOf _ -> false) columns
 
 (* all columns from tables, without duplicates *)
-(* FIXME check type of duplicates *)
-let make_unique = List.unique ~cmp:Schema.Source.Attr.(fun a1 a2 -> a1.attr.name = a2.attr.name && a1.attr.name <> "")
-
-let all_columns  = make_unique $ Schema.cross_all
+let make_unique =
+  List.unique ~cmp:(fun a1 a2 ->
+      (* Check if alias has same name as column *)
+      (List.length a2.Schema.Source.Attr.sources = 0
+      (* Check if columns are from the same table (source)  *)
+      || (List.length a1.sources = List.length a2.sources && List.for_all2 ( = ) a1.sources a2.sources))
```

Comment:
```suggestion
     (* Check if columns are from the same table (source)  *)
      (a2.Schema.Source.Attr.sources = [] || a1.sources = a2.sources)
```
less confusing indentation and simplify

Reply:
why is a2.sources=[] check is needed? it means if a2.sources is empty and a1.sources is not empty then result will be true, ie equality is not symmetric? :thinking:

Reply:
```sql 
SELECT COUNT(column_a) as column_a FROM test21 WHERE column_a = @column_a
```
This is when we alias a column to the same name as a column in selected tables

Reply:
put this in code comment
i am still not convinced why `make_unique a b` is allowed to be different from `make_unique b a`

Reply:
> i am still not convinced why make_unique a b is allowed to be different from make_unique b a
Fixed